resolwe/flow/tests/test_models.py

```diff
@@ -292,6 +300,32 @@ def test_save_storage(self):
         storage = Storage.objects.first()
         self.assertEqual(data.output['json_field'], storage.pk)
 
+    def test_save_storage_file(self):
+        """File is loaded and saved to storage"""
+        data = Data.objects.create(
+            name='Test data',
+            contributor=self.user,
+            process=self.proc,
+        )
+
+        data.output = {'json_field': 'json.txt'}
+        data.status = Data.STATUS_DONE
+
+        isfile_mock = MagicMock(return_value=True)
+        if six.PY3:
```

Comment:
I think a nicer and more general solution is to use `io.BytesIO`, open the stream with `open` and use the `t` mode to open the stream in text mode. Then, one can just use `json.dump` rather than `json.dumps`.

The thing is that `json.dumps` will serialize the given object to a string, which creates problems when trying to write Python 2 & 3 compatible code.

My proposed solution would be:
```python
json_file = io.BytesIO()
with open(json_file, mode='wt') as f:
    json.dump({'foo': 'bar'}, f)
# set seek position of the binary stream object back to 0
json_file.seek(0)

isfile_mock = MagicMock(return_value=True)
open_mock = MagicMock(return_value=json_file)
with patch.object(os.path, 'isfile', isfile_mock):
    with patch.object(builtins, 'open', open_mock):
        data.save()
```

I used a similar thing in [`test_test_framework`'s tests](https://github.com/genialis/resolwe/blob/master/resolwe/flow/tests/test_test_framework.py#L64-L89).

Note that I would also drop `_mock` from the `json_file_mock` variable name since the variable is not a Mock object but rather a real in-memory file.